marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/customStep/updateCustomStep.sjs

```diff
@@ -0,0 +1,47 @@
+/**
+ Copyright (c) 2021 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-custom", "execute");
+
+const Artifacts = require('/data-hub/5/artifacts/core.sjs');
+const consts = require("/data-hub/5/impl/consts.sjs");
+
+var stepProperties;
+
+stepProperties = stepProperties.toObject();
+const stepName = stepProperties.name;
+
+// check step already exists
+let existingStep = fn.head(cts.search(cts.andQuery([
```

Comment:
Should be able to use Artifacts.getArtifact("custom", name) here

Reply:
Created a new function called artifactExists in core.sjs.  Using that instead.

Reply:
I don't think that function is needed. You can just do this 
```
let existingStep = Artifacts.getArtifact("custom", stepName)
if(!existingStep){
//throw error
} 
```

Reply:
Artifacts.getArtifact will throw its own 404 if the step doesn't already exist, and personally I don't think that error message is all that helpful (it just says this step doesn't exist).  The artifactExists function will return true/false based on whether the step exists or not.  That way the error can be thrown in updateCustomStep.sjs

Reply:
I think the 404 that Artifacts.getArtifact throws is sufficient. Users are likely not going to hit this scenario as they can only edit a custom step from HC if it is present.